### DISCUSSION TITLE
🎨 Pixel Prime Development Feedback: Team Input Needed! 🎮

### DISCUSSION THREAD

**Discussion by Maxiboy441** [Maxiboy441/schoolproject-3#156] [Q&A]
Hey Team Pixel Prime! 👋

As you all know, we're about one week away from the project deadline for our school assignment. I thought it would be a great time to gather some feedback from everyone on how things have been going so far. This is a fantastic opportunity to reflect on our experience as developers, share thoughts on the project, and discuss both what we did well and what we could improve.

Here's what I'd like to get your input on:

---

### **1. Development Process Feedback** 🛠️
- What aspects of our development process went smoothly?
- What do you think could have been done better or more efficiently?
- How was the collaboration between us as a team of 5 developers?
- Did we have any blockers or issues that slowed down development? If so, how did we handle them?

### **2. Pixel Prime Platform Feedback** 🎮
- What do you like most about the platform we’ve built so far? (Feel free to mention any features, design elements, or overall user experience)
- Are there any parts of the platform that could use more improvement or polish before the final submission?
- Do you think the platform meets the initial vision we set out with, or did it evolve into something different?

### **3. Personal Development** 🌱
- How do you feel you’ve grown as a developer over the course of this project?
- Did this project challenge you in ways that helped you improve any specific skills (coding, design, problem-solving, etc.)?
- Are there areas where you felt you could have contributed more or learned more? What would have helped you get there?

### **4. Team Organization & Communication** 🧠
- How did you feel about the organization and structure of our project workflow (e.g., task distribution, deadlines, code reviews, etc.)?
- What did you think about our communication throughout the project? Were there ways we could have improved how we shared updates, discussed issues, or resolved challenges?
- Is there any feedback on how we managed our sprints, tasks, or the tools we used (GitHub,etc.)?

---

Feel free to answer as many questions as you’d like or add any additional thoughts that come to mind. Your input is really important to me as we look back on the project and, more importantly, look forward to improving our individual work and growth for the future.

Thanks for taking the time to share your thoughts, and let's push strong for this last week to finish Pixel Prime on a high note! 💪

Best,  
@Maxiboy441

**Comment by Maxiboy441** [comment]
<h1>1. Development Process Feedback 🛠️</h1>
<h5>What aspects of our development process went smoothly?</h5>
Our concept phase was really indept and focused on everyones skill level. We produced a unit workflow which everyone learned, which was awesome. Tickets where taken and done in like no time, after a few weeks we had out first version live. Also the support of everyone was really good.
<h5>What do you think could have been done better or more efficiently?</h5>
The only points where we could improve would be I guess that 1 person did not much and 1 did nothing. Maybe we could also improve by learning the workflow more, but thats would be nitpick.
<h5>How was the collaboration between us as a team of 5 developers?</h5>
@feliperamoss and @mehrb98 did a awesome job of supporting each other, myself included. We three all learned from eachother and helped us with dificult tasks. There where even calls in the middle of the night between some of us for support.
<h5>Did we have any blockers or issues that slowed down development? If so, how did we handle them?</h5>
The only time we had a blocker was at the start with the User Model which was supposed to be done with a package by @mehrb98. He worked his ass of to try it, but in the end we had to build it ourselves which took extra time.

<h1>2. Pixel Prime Platform Feedback 🎮</h1>
<h5>What do you like most about the platform we’ve built so far? (Feel free to mention any features, design elements, or overall user experience)</h5>
I love the overall design, specially the animations on the user-page and the user docu. Im also really proud that we handled the recommendation service ai and the actor feature, both where really work intensive.
<h5>Are there any parts of the platform that could use more improvement or polish before the final submission?</h5>
Bugs, we have them and we constantly trying to fix them but there a still a few.
<h5>Do you think the platform meets the initial vision we set out with, or did it evolve into something different?</h5>
Yes, we even overshoot our expectations by a loot.


<h1>3. Personal Development 🌱</h1>
<h5>How do you feel you’ve grown as a developer over the course of this project?</h5>
I learned C# much more and now even liking it. I also learned more about github, which has so many features that I wasnt even able to try all of them.
<h5>Did this project challenge you in ways that helped you improve any specific skills (coding, design, problem-solving, etc.)?</h5>
Sure, specially management and devops wise I had new challages to solve.
<h5>Are there areas where you felt you could have contributed more or learned more? What would have helped you get there?</h5>
I could have coded more, but I dont know in what time tbh. I spend so much time on management, devops etc that I did not as much tickets as I would've liked, but still far from none I guess.

<h1>4. Team Organization & Communication 🧠</h1>
<h5>How did you feel about the organization and structure of our project workflow (e.g., task distribution, deadlines, code reviews, etc.)?</h5>
The orga was really good. Tasks distribution was hard at the start because of us not knowing each other, but went well really fast except of the 2 people which rarely/never took or done tasks. Reviews where really good, pickie but relatable, and deadlines where never an issue.
<h5>What did you think about our communication throughout the project? Were there ways we could have improved how we shared updates, discussed issues, or resolved challenges?</h5>
Communication was a 10/10, everything was done kindly and either via whatsapp, github or in person. Everything was straight forward and fast.
<h5>Is there any feedback on how we managed our sprints, tasks, or the tools we used (GitHub,etc.)?</h5>
Was fine, liked the MVP and Imrovement approach. With that we could split the features in the necessary and would like to have tasks.


All in all im really happy with how it went and the result.

**Comment by feliperamoss** [comment]
**1. Development Process Feedback 🛠️**
What aspects of our development process went smoothly?
Concept was done pretty, we all worked on the concept together doing a Brainstorm. Overall, the whole development went well.

What do you think could have been done better or more efficiently?
Maybe we should have been  to realize that we could not use the user management package and go for another plan.

How was the collaboration between us as a team of 5 developers?
We worked in sync, helping each other.

Did we have any blockers or issues that slowed down development? If so, how did we handle them?
User Management feature was a blocker, maybe we should decided faster to go for another plan.

**2. Pixel Prime Platform Feedback 🎮**
What do you like most about the platform we’ve built so far? (Feel free to mention any features, design elements, or overall user experience)
I like the my space page where all the favorites, watchlist, recommendations, reviews and ratings of the user are displayed. I also like the recommendation service.

Are there any parts of the platform that could use more improvement or polish before the final submission?
I think the home page could have improvements in the design.

Do you think the platform meets the initial vision we set out with, or did it evolve into something different?
We did as we planned. And also added new features like showing information about actors.

**3. Personal Development 🌱**
How do you feel you’ve grown as a developer over the course of this project?
I feel much more confident in developing an application in C#.

**4. Team Organization & Communication 🧠**
How did you feel about the organization and structure of our project workflow (e.g., task distribution, deadlines, code reviews, etc.)?
We supported where we could, according to the skills of each one of the group.

What did you think about our communication throughout the project? Were there ways we could have improved how we shared updates, discussed issues, or resolved challenges?
I think communication was quite good, we had a whatsapp group and also added comments on Tickets and Pull Requests.

Is there any feedback on how we managed our sprints, tasks, or the tools we used (GitHub,etc.)?
It is really cool the automation done for the project in GitHub, project was managed really well and organised. Tickets definition accordingly, the tickets had also information about for example size, time ( for example estimation), area( example: Backend, Frontend), so it was very well managed.

**Comment by Mariuswenzelits** [comment]
1. Development Process Feedback 🛠️
What aspects of our development process went smoothly?
I feel like we did extremly well building the first foundation.

What do you think could have been done better or more efficiently?
I think the usage of the School laptops made it a bit harder for me personally to be as efficient as possible.

How was the collaboration between us as a team of 5 developers?
When you asked for help or had any Questions they were always answered.

2. Pixel Prime Platform Feedback 🎮
What do you like most about the platform we’ve built so far?
I love how smooth everything works, like the favorites / watchlist page is really good.

Are there any parts of the platform that could use more improvement or polish before the final submission?
I really dont like the way the homepage sliders work, when you go left or right, it looks a bit weird.

Do you think the platform meets the initial vision we set out with, or did it evolve into something different?
I think we have met our goal, and shot beyond it a little bit (adding extra features).

3. Personal Development 🌱
How do you feel you’ve grown as a developer over the course of this project?
I have learned a lot about Databases and Project structures.

4. Team Organization & Communication 🧠
How did you feel about the organization and structure of our project workflow (e.g., task distribution, deadlines, code reviews, etc.)?
I think the organisation was one of our strong point during this project.

What did you think about our communication throughout the project? Were there ways we could have improved how we shared updates, discussed issues, or resolved challenges?
I think our communication was very good tbh.

Is there any feedback on how we managed our sprints, tasks, or the tools we used (GitHub,etc.)?
I think it was very well managed and i loved how much work went into our github, the only thing that i would have personally liked was maybe more smaller sprints.

**Comment by mehrb98** [comment]
### 1. Development Process Feedback 🛠️

**• What aspects of our development process went smoothly?**

The project concept was well-planned, which made development much easier. Having a clear roadmap allowed us to approach tasks efficiently, and overall, I found the process quite smooth.

**• What do you think could have been done better or more efficiently?**

We could have tested the user management system with Docker before starting development. This would have helped us identify potential issues early on and save time later.

**• How was the collaboration between us as a team of 5 developers?**

Communication was effective, particularly with Philip and Max, who worked hard and explained things thoroughly. Their support was invaluable to me throughout the project.

**• Did we have any blockers or issues that slowed down development? If so, how did we handle them?**

The biggest blocker was user management. It caused significant delays as I spent a lot of time setting it up, and in the end, it didn’t integrate well with Docker. We worked through it, but it cost us time.

### 2. Pixel Prime Platform Feedback 🎮

**• What do you like most about the platform we’ve built so far?**

I’m really happy with the My Space, Movie Page, and Top Rated Movies and Series sections. They turned out well and align with my expectations.

**• Are there any parts of the platform that could use more improvement or polish before the final submission?**

If there are no major bugs, I think we’re in good shape. Everything looks well-polished from my perspective.

**• Do you think the platform meets the initial vision we set out with, or did it evolve into something different?**

I think it’s quite similar to what we originally envisioned but even better than we planned. It’s exceeded my expectations.

### 3. Personal Development 🌱

**• How do you feel you’ve grown as a developer over the course of this project?**

Absolutely. This project helped me improve my understanding of GitHub, explore new C# functions and methods, and get comfortable using Docker.

**• Did this project challenge you in ways that helped you improve any specific skills?**

Yes, I was definitely challenged, particularly with coding and learning how to work with Docker. It pushed me to enhance my skills and problem-solving abilities.

**• Are there areas where you felt you could have contributed more or learned more? What would have helped you get there?**

I feel that I contributed well, but more time with Docker and user management integration would have helped me master those areas even further.

### 4. Team Organization & Communication 🧠

**• How did you feel about the organization and structure of our project workflow?**

The project was well-organized and the planning was clear and understandable. I appreciated the way tasks were distributed and deadlines were managed.

**• What did you think about our communication throughout the project? Were there ways we could have improved how we shared updates, discussed issues, or resolved challenges?**

Our communication was generally good, but I think it would have been helpful to exchange more personal experiences before we started. Understanding who was skilled with which technologies would have allowed us to better leverage our strengths. For example,  @Maxiboy441 helped me a lot with GitHub, which sped up my progress.

**• Is there any feedback on how we managed our sprints, tasks, or the tools we used (GitHub, etc.)?**

I felt comfortable with the tools and the workflow, but perhaps more team knowledge-sharing before starting the project would have streamlined things a bit more.

**Comment by AmirHassanMojahed** [comment]
Development Process Feedback 🛠️

What went well in our development process?
We did a fantastic job laying a strong groundwork for the project, setting a solid direction from the start.

What could have been improved or made more efficient?
It would have been easier for me to work more efficiently if I had access to better tools. The school laptops sometimes limited my productivity.

How was the collaboration between the five of us as a team?
Whenever I needed assistance or had questions, I always got helpful and timely responses from the team.
Pixel Prime Platform Feedback 🎮

What do you like most about the platform we’ve built so far?
I really appreciate how seamlessly everything works, especially the favorites and watchlist features, which are very well implemented.

Are there any parts of the platform that need improvement or polishing before submission?
The homepage sliders don’t feel very smooth when scrolling left or right; it could be improved to look and function better.

Do you feel the platform aligns with the original vision, or has it evolved into something different?
I think we’ve hit our targets and even gone beyond them by incorporating additional features.
Personal Development 🌱
How have you grown as a developer throughout this project?
I’ve significantly expanded my knowledge of databases and project structure throughout this experience.
Team Organization & Communication 🧠
What are your thoughts on the organization and structure of our workflow (task distribution, deadlines, code reviews, etc.)?
I think we had great organization, which made the project run smoothly.

How did you feel about our communication during the project? Were there ways we could have improved in terms of sharing updates, discussing issues, or resolving challenges?
I was very happy with how we communicated. Everything was clear and effective.

Any feedback on how we handled sprints, tasks, or the tools we used (like GitHub)?
We managed the project tools, like GitHub, really well. One thing I would have liked is having more frequent but shorter sprints to keep up the pace.
